app/service/parsing_svc.py

```diff
@@ -1,14 +1,48 @@
-import plugins.stockpile.parsers.standard as parsers
-import plugins.stockpile.parsers.mimikatz as mimikatz_parser
+import glob
 from base64 import b64decode
+from pydoc import locate
 
 from app.service.base_service import BaseService
 
 
 class ParsingService(BaseService):
 
     def __init__(self):
+        self.parsers = dict()
         self.log = self.add_service('parsing_svc', self)
+        
+    async def load_parser(self, p_id, name, parser):
+        """
+        Load the specified parser plugin
+        :param p_id:  the id of the yaml file that defines the parser
+        :param name:  the unique name of the parser
+        :param parser:  an instance of the parser class 
+        :return: None
+        """        
+        if name in self.parsers:
+            self.log.warning('Duplicate parser name detected:  %s:%s' % (p_id, name))
+        else:
+            self.parsers[name] = parser       
+        
+    async def load_parsers(self, directory):
+        """
+        Load all parser plugins
+        :param directory:
+        :return: None
+        """
+        for filename in glob.iglob('%s/*.yml' % directory, recursive=False):
+            for entries in self.strip_yml(filename):
+                parser = locate(entries['class'])
+                if parser:
+                    parser = parser(self.log)
+                else:
+                    self.log.warning('Unable to find parser class %s' % entries['class'])
+                    next
+                if isinstance(entries['name'], str):                    
```

Comment:
try to avoid type checking

Reply:
How do you recommend I do this in a more pythonic way that avoids type checking unless your recommendation here is to catch the TypeError?